client/messaging/message-list.tsx

```diff
@@ -49,11 +60,25 @@ const handleUnknown = (msg: Message) => {
 const PureMessageList = React.memo<PureMessageListProps>(({ messages, renderMessage }) => {
   return (
     <Messages>
-      {messages.map(m => {
+      {messages.map((m, index, array) => {
         // NOTE(2Pac): We only handle common messages here, e.g. text message. All other types of
         // messages are handled by calling the `renderMessage` function which should be supplied by
         // each service if they have any special messages to handle.
-        return renderCommonMessage(m) ?? (renderMessage ?? handleUnknown)(m)
+        const messageLayout = renderCommonMessage(m) ?? (renderMessage ?? handleUnknown)(m)
+
+        const nextMessage = array.get(index + 1)
+        if (!nextMessage) {
+          return messageLayout
+        }
+
+        const newDayMessageRecord = new NewDayMessageRecord({
+          id: cuid(),
```

Comment:
Doing this will mean these messages will never be the same on each render, you should prefer something deterministic instead (e.g. the timestamp concatenated with some string unique to new day messages)